components/hostobjects/org.jaggeryjs.hostobjects.file/src/main/java/org/jaggeryjs/hostobjects/file/FileHostObject.java

```diff
@@ -369,4 +369,196 @@ public static Object jsFunction_listFiles(Context cx, Scriptable thisObj, Object
         return cx.newArray(thisObj, fhol.toArray());
     }
 
+    /**
+     * To unzip a zip file
+     *
+     * @param cx      Context
+     * @param thisObj FileHostObject to be unzipped
+     * @param args    Path to unzip the zip file
+     * @param funObj  Function Object
+     * @throws ScriptException
+     */
+    public static boolean jsFunction_unZip(Context cx, Scriptable thisObj, Object[] args, Function funObj)
+            throws ScriptException, IOException {
+        String functionName = "unZip";
+        int argsCount = args.length;
+        if (argsCount != 1) {
+            HostObjectUtil.invalidNumberOfArgs(hostObjectName, functionName, argsCount, false);
+        }
+        FileHostObject fho = (FileHostObject) thisObj;
+        ZipInputStream zin = null;
+        if (fho.file.isExist()) {
+            JaggeryContext context = (JaggeryContext) RhinoEngine.getContextProperty(EngineConstants.JAGGERY_CONTEXT);
+            Object obj = context.getProperty(JAVASCRIPT_FILE_MANAGER);
+            if (obj instanceof JavaScriptFileManager) {
+                fho.manager = (JavaScriptFileManager) obj;
+            } else {
+                fho.manager = new JavaScriptFileManagerImpl();
+            }
+            File zipfile = new File(fho.manager.getFile(fho.file.getPath()).getAbsolutePath());
+            File outdir = new File(fho.manager.getDirectoryPath(args[0].toString()));
+            outdir.getParentFile().mkdirs();
+            outdir.mkdir();
+            try {
+                zin = new ZipInputStream(new FileInputStream(zipfile));
+                ZipEntry entry;
+                String name, dir;
+                byte[] buffer = new byte[1024];
+                while ((entry = zin.getNextEntry()) != null) {
+                    name = entry.getName();
+                    if (entry.isDirectory()) {
+                        mkdirs(outdir, name);
+                        continue;
+                    }
+                    int hasParentDirs = name.lastIndexOf(File.separatorChar);
+                    dir = (hasParentDirs == -1) ? null : name.substring(0, hasParentDirs);
+                    if (dir != null) {
+                        mkdirs(outdir, dir);
+                    }
+                    BufferedOutputStream out = new BufferedOutputStream(new FileOutputStream(new File(outdir, name)));
+                    try {
```

Comment:
Not necessarily you need to have two try blocks, this can be done with only single try block as the inner try block doesn't do anything additional.